course/models/models.py

```diff
@@ -342,12 +342,12 @@ def save(self, *args, **kwargs):
         if not self.finalized:
             self.calculate_grade(commit=False)
 
-        if not self.in_progress and self.is_correct or self.is_partially_correct:
+        if (not self.in_progress and self.is_correct or self.is_partially_correct) or self.question.event.is_exam:
```

Comment:
I'm a little worried that you might have used question.event without checking if it's None in other places. can you please check all the usage of question.event and make sure they are used safely?

Reply:
For sure, good catch.